qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+        - What is the formatting for the results when measuring a subset of qubits? E.g. if
+          you measure qubits [0, 2, 3], do you get counts for 3-bit strings out?
+        - Add validation that only qubits present in the model are being measured. This may
+          happen automatically as the measurement code will fail, but would be good to raise
+          a proper error.
+        - How to handle binning of higher level measurements? E.g. should everything above 0
+          count as 1?
+        - What to do with memory and register slots?
+
+        To test:
+        - Measuring a subset of qubits when more than one present
+        - If t_span=[0, 0] it seems that NaNs appear for JAX simulation - should solve this.
+        - Normalizing the state.
+        """
+
+        if validate:
+            _validate_run_input(run_input)
+
+        if solver_options is None:
+            solver_options = {}
+
+        if y0 is None:
+            y0 = Statevector(self._dressed_states[:, 0])
+
+        # to do: add handling of circuits
+        schedules = _to_schedule_list(run_input)
+
+        # get the acquires instructions and simulation times
+        t_span = []
+        measurement_subsystems_list = []
+        for schedule in schedules:
+            schedule_acquires = []
+            schedule_acquire_times = []
+            for start_time, inst in schedule.instructions:
+                if isinstance(inst, pulse.Acquire):
+                    schedule_acquires.append(inst)
+                    schedule_acquire_times.append(start_time)
+
+            # maybe need to validate more here
+            _validate_acquires(schedule_acquire_times, schedule_acquires)
+
+            t_span.append([0, schedule_acquire_times[0]])
+            measurement_subsystems_list.append([inst.channel.index for inst in schedule_acquires])
+            measurement_subsystems_list[-1].sort()
+
+
+        # Build and submit job
+        job_id = str(uuid.uuid4())
+        dynamics_job = DynamicsJob(
+            backend=self,
+            job_id=job_id,
+            fn=self._run,
+            fn_kwargs={
+                't_span': t_span, 'y0': y0, 'schedules': schedules,
+                'solver_options': solver_options,
+                'measurement_subsystems_list': measurement_subsystems_list,
+                'normalize_states': normalize_states,
+                'shots': shots
+            }
+        )
+        dynamics_job.submit()
+
+        return dynamics_job
+
+    def _run(
+        self,
+        job_id,
+        t_span,
+        y0,
+        schedules,
+        solver_options,
+        normalize_states,
+        measurement_subsystems_list,
+        shots
+    ) -> Result:
+        """Not sure here what the right delineation of arguments is to put in _run.
+        This feels somewhat hacky/arbitrary.
+        """
+        start = time.time()
+
+        # map measurement subsystems from labels to correct index
+        if self.subsystem_labels:
```

Comment:
This looks like something that can be run and cached once when the relevent option value is set or changed, not evaluated every `_run` call.